### PR TITLE
add MultiPolygon wrapper around OGRMultiPolygon

### PR DESCRIPTION
**Pull request by susterdan**

### CODE REVIEW

File: geo/csconvertor.hpp [hunk 1]
```diff
@@ -162,6 +168,14 @@ inline math::Points3 CsConvertor::operator()(const math::Points3 &p) const
     return out;
 }
 
+#ifdef GEO_HAS_GDAL
+inline OGRPoint CsConvertor::operator()(const OGRPoint& p) const
```
Comment (vaclavblazek) [inline geo/csconvertor.hpp]: What if `OGRPoint` has Z-component?

Comment (susterdan) [reply geo/csconvertor.hpp]: Right. Move it out to polygons loading with 2D restriction comment.

File: geo/ogrpolygon.cpp [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        LOGTHROW(err3, std::runtime_error)
+            << "Failed to open dataset " << path
+            << ", errno: " << ::CPLGetLastErrorNo()
+            << ", error msg: " << ::CPLGetLastErrorMsg();
+    }
+    return dataset;
+}
+
+MultiPolygon loadPolygons(const fs::path& path,
+                          boost::optional<SrsDefinition> targetSrs)
```
Comment (vaclavblazek) [inline geo/ogrpolygon.cpp]: Oh, pls, switch to `std::optional` (from `#include <optional>`).
NB: `boost::none` -> `std::nullopt`

File: geo/ogrpolygon.cpp [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+    OGRPolygon polygon;
+    polygon.addRing(&ring);
+    return polygons_.Overlaps(&polygon);
+}
+
+bool MultiPolygon::overlaps(const math::Points2& other) const
+{
+    if (other.size() < 3) return false;
+    OGRLinearRing ring;
+    ring.addPoint(other[0][0], other[0][1]);
```
Comment (zap150) [inline geo/ogrpolygon.cpp]: Why don't we construct the ring from possibly all points in other?

File: geo/ogrpolygon.cpp [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+
+    polygon.addRing(&ring);
+    polygons.addGeometry(&polygon);
+    return { polygons };
+}
+
+GDALDatasetUniquePtr openDataset(const fs::path& path)
+{
+    GDALDatasetUniquePtr dataset {
+        GDALDataset::Open(reinterpret_cast<const char*>(path.c_str()), GDAL_OF_VECTOR | GDAL_OF_READONLY)
```
Comment (zap150) [inline geo/ogrpolygon.cpp]: @pato-melowntech why do we need the cast for Win?

Comment (pato-melowntech) [reply geo/ogrpolygon.cpp]: This should work, reinterpret cast does not make much sense.
```suggestion
        GDALDataset::Open(path.generic_string().c_str(), GDAL_OF_VECTOR | GDAL_OF_READONLY)
```

**Review by zap150 — APPROVED** [review]

**Pull request merged**